src/treelearner/gpu_tree_learner.cpp

```diff
@@ -599,7 +599,7 @@ void GPUTreeLearner::BuildGPUKernels() {
     }
     histogram_kernels_[i] = program.create_kernel(kernel_name_);
 
-    // kernel with all features enabled, with elimited branches
+    // kernel with all features enabled, with eliminated branches
```

Comment:
I don't understand this part of the code, so I'm not actually sure if this was supposed to be "limited" or "eliminated".

I can't really tell based on the other code around this.

@shiyu1994 or @StrikerRUS could you help with a suggestion for this one?

Reply:
I am totally not an expert on this code, but in `\src\treelearner\ocl\histogram16.cl` on lines 346 there is a related statement. 

```
#if ENABLE_ALL_FEATURES == 1
    // hopefully the compiler will propogate the constants and eliminate all branches
```

So I think "eliminated" is correct

Reply:
ah, nice! Ok yes that makes sense, thank you